grails-app/assets/javascripts/csiBenchmark/csiBenchmarkChart.js

```diff
@@ -85,17 +86,18 @@ OpenSpeedMonitor.ChartModules.CsiBenchmarkChart = (function (chartIdentifier) {
             });
 
         // enter
-        var barContainer = bars.enter().append("g")
-            .attr("transform", function (d) {
-                return "translate(" + xScale(d.name) + "," + yScale(d.value) + ")";
-            })
+        var barContainer = bars.enter()
+            .append("g")
             .attr("class", "d3chart-bar-container");
         barContainer.append("rect")
             .attr("class", "d3chart-bar")
             .attr("height", function (d) {
                 return height - yScale(d.value);
             })
             .attr("width", barWidth)
+            .attr("transform", function (d) {
```

Comment:
Note that this is not necessary if the calculation and assignment of values is already present in the update function. Same goes for all other dynamic assignments of values.

Reply:
Likewise, as was done with the width, it is best to assign the width or x and y attributes in the update section.

Reply:
Sorry for all the redundancy. Should be more readable now and without update functionality in enter section.